vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java

```diff
@@ -167,4 +186,74 @@ public Page<ISlide> findByNameOrDescription(Pageable requestedPage, String searc
 
         return slideRepo.findDistinctByNameContainingOrDescriptionContaining(requestedPage, searchText,searchText);
     }
+
+    @Override
+    public void updateNameAndDescription(ISlide slide, SlideForm slideForm) {
+        slide.setName(slideForm.getDefaultName().getText());
+        slide.setDescription(slideForm.getDefaultDescription().getText());
+        addSlideName(slide,slideForm.getDefaultName());
+        addSlideDescription(slide,slideForm.getDefaultDescription());
+        
+        for(LocalizedTextForm title:slideForm.getNames()) {        
+            addSlideName(slide,title);
+        }
+        for(LocalizedTextForm text:slideForm.getDescriptions()) {
+            addSlideDescription(slide,text);
+        }
+    }
+    
+    /**
+     * Adds name to slideNames List
+     * @param slide
+     * @param names
+     */
+    @Override
+    public void addSlideName(ISlide slide, LocalizedTextForm name) {
+        if(name!=null && !StringUtils.isEmpty(name.getText())) {
+            LocalizedText localizedText = localizedTextRepo.findById(name.getLocalisedTextId()).orElse(null);
+            if(localizedText != null) {
+                localizedText.setText(name.getText());
+            } else {
+                ExhibitionLanguage exhibitionLanguage = exhibitionLanguageRepository.findById(name.getExhibitionLanguageId())
+                        .orElse(null);
+                if(exhibitionLanguage != null) {
+                    localizedText = new LocalizedText(exhibitionLanguage, name.getText());
+                    slide.getSlideNames().add(localizedText);
+                }
+            }
+        }
+    }
+    
+    /**
+     * Adds description to slideDescription list.
+     * @param slide
+     * @param descriptions
+     */
+    @Override
+    public void addSlideDescription(ISlide slide, LocalizedTextForm description) {
+        if(description!=null && !StringUtils.isEmpty(description.getText())) {
+            LocalizedText localizedText = localizedTextRepo.findById(description.getLocalisedTextId()).orElse(null);
+            if(localizedText != null) {
+                localizedText.setText(description.getText());
+            } else {
+                ExhibitionLanguage exhibitionLanguage = exhibitionLanguageRepository.findById(description.getExhibitionLanguageId())
+                        .orElse(null);
+                if(exhibitionLanguage != null) {
+                    LocalizedText newLocalizedText = new LocalizedText(exhibitionLanguage, description.getText());
+                    slide.getSlideDescriptions().add(newLocalizedText);
+                }
+            }
+        }
```

Comment:
same comments as in the other class

Reply:
The  method "updateNameAndDescription" in SlideManager uses addSlideDescription.

Reply:
duplicate code